polartools/hkl_utils.py

```diff
@@ -1002,3 +1099,83 @@ def setlat(a=None, b=None, c=None, alpha=None, beta=None, gamma=None):
             sample._orientation_reflections[1],
         )
         _geom_.forward(1, 0, 0)
+
+
+def write_config(method="File", overwrite=False):
+    config = DiffractometerConfiguration(_geom_)
+    # config_file = pathlib.Path("diffractometer-config.json")
+    settings = config.export("json")
+    if pathlib.Path("diffractometer-config.json").exists():
+        if not overwrite:
+            value = input("Overwrite existing configuration file (y/[n])? ")
+            if value == "y":
+                overwrite = True
+        if overwrite:
+            if method == "File":
+                print("Writing configuration file.")
+                with open("diffractometer-config.json", "w") as f:
+                    f.write(settings)
+
+
+def read_config(method="File"):
+    config = DiffractometerConfiguration(_geom_)
+    if pathlib.Path("diffractometer-config.json").exists():
+        if method == "File":
+            print(
+                "Read configuration file '{}'.".format(
+                    "diffractometer-config.json"
+                )
+            )
+            method = input("Method ([o]verwrite/[a]ppend)? ")
+            if method == "a":
+                config.restore(
+                    pathlib.Path("diffractometer-config.json"), clear=False
+                )
+            elif method == "o":
+                config.restore(
+                    pathlib.Path("diffractometer-config.json"), clear=True
+                )
+
+
+from inspect import getmembers, isfunction
```

Comment:
The error in flake8 is a style one and comes from these imports in the middle of the file. You need to move them to the top.